source/MaterialXTest/Definition.cpp

```diff
@@ -0,0 +1,52 @@
+//
+// TM & (c) 2017 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXTest/Catch/catch.hpp>
+
+#include <MaterialXCore/Definition.h>
+#include <MaterialXCore/Document.h>
+#include <MaterialXFormat/XmlIo.h>
+
+namespace mx = MaterialX;
+
+TEST_CASE("Implementation Check", "[definition]")
+{
+    // Create document and load in standard definitions and reference implementation
+    //
+    mx::DocumentPtr doc = mx::createDocument();
+
+    std::vector<std::string> filenames =
+    {
+        "documents/Libraries/mx_stdlib_defs.mtlx",
+        "documents/Libraries/mx_stdlib_impl_osl.mtlx",
+    };
+    for (const std::string& filename : filenames)
+    {
+        mx::readFromXmlFile(doc, filename);
+    }
+
+    // Check that there is one implementation per nodedef.
```

Comment:
This test case is similar to existing logic in MaterialXTest/XmlIo.cpp, which verifies that each referenced node has both a NodeDef and an Implementation.  Perhaps it would be better to extend that logic, so that it additionally verifies that each NodeDef in the standard library (with the noted exceptions) has an Implementation?